044-Stable-identities-for-Kafka-Connect-worker-nodes.md

```diff
@@ -0,0 +1,176 @@
+Since all Pods will share the same configuration, there is no need to support things such as scaling down pods from the middle or beginning of the sequence.
+
+### Feature Gate
+
+The stable identities for the Kafka Connect worker nodes will be introduced gradually using a Feature Gate.
+There will be only one Feature Gate for both Kafka Connect and Kafka Mirror Maker 2.
+The new Feature gate will be named `StableConnectIdentities`.
+To enable this feature gate, the `UseStrimziPodSets` feature gate would need to be enabled as well.
+The following table shows the expected graduation of the `StableConnectIdentities` feature gate:
+
+| Phase | Strimzi versions       | Default state                                          |
+|:------|:-----------------------|:-------------------------------------------------------|
+| Alpha | 0.34, 0.35             | Disabled by default                                    |
+| Beta  | 0.36, 0.37             | Enabled by default                                     |
+| GA    | 0.38 and newer         | Enabled by default (without possibility to disable it) |
+
+### Migration from Kubernetes Deployments to StrimziPodSets
+
+The migration from Kubernetes Deployments to StrimziPodSets will be used in the following situations:
+* Upgrading from Strimzi versions which does not use StrimziPodSets for Kafka Connect to a Strimzi version using StrimziPodSets for Kafka Connect
+    * This includes upgrades from Strimzi versions before introduction of the `StableConnectIdentities` feature gate
+    * Or upgrades from version with disabled `StableConnectIdentities` feature gate to version where it is enabled
+* Enabling of the `StableConnectIdentities` feature gate in Strimzi version which supports it
+
+The migration will happen automatically and will use the following steps:
+* If the user is using the `RollingUpdate` deployment strategy, Strimzi will
+    1) Create an empty StrimziPodSet without any Pods
+    2) Add new Kafka Connect Pod to the StrimziPodSet
+    3) Wait for the new Pod to become ready
+    4) Scale-down the Deployment by one replica to remove one Kafka Connect Pod from the Deployment
+    5) ... repeat steps 2-4 until the StrimziPodSet has the desired number of Pods and the Deployment has 0 replicas
+    6) Delete the Deployment
+* If the user is using the `Recreate` deployment strategy, Strimzi will
+    1) Create an empty StrimziPodSet without any Pods
+    2) Scale-down the Deployment by one replica to remove one Kafka Connect Pod from the Deployment
+    3) Add new Kafka Connect Pod to the StrimziPodSet
+    4) Wait for the new Pod to become ready
+    5) ... repeat steps 2-4 until the StrimziPodSet has the desired number of Pods and the Deployment has 0 replicas
+    6) Delete the Deployment
+
+Doing the things in the right oder is important because otherwise it might happen that there will be no space in the cluster to start the new Pod and the migration might get stuck.
```

Comment:
```suggestion
Doing the things in the right order is important because otherwise it might happen that there will be no space in the cluster to start the new Pod and the migration might get stuck.
```